lib/luahid.c

```diff
@@ -93,6 +94,84 @@ static const struct hid_device_id *luahid_setidtable(lua_State *L, int idx)
 	return user_table;
 }
 
+static int luahid_probe(struct hid_device *hdev, const struct hid_device_id *id);
+
+static void luahid_pushdevid(lua_State *L, const struct hid_device_id *id) {
+	lua_newtable(L);
+	lua_pushinteger(L, id->bus); lua_setfield(L, -2, "bus");
+	lua_pushinteger(L, id->group); lua_setfield(L, -2, "group");
+	lua_pushinteger(L, id->vendor); lua_setfield(L, -2, "vendor");
+	lua_pushinteger(L, id->product); lua_setfield(L, -2, "product");
+	lua_pushinteger(L, id->driver_data); lua_setfield(L, -2, "driver_data");
+}
+
+static int luahid_probe_handler(lua_State *L, luahid_t *hid, struct hid_device *hdev, const struct hid_device_id *id) {
+	int ret = 0;
+
+	if (lunatik_getregistry(L, hid) != LUA_TTABLE) {
+		pr_err("Could not find the hid table");
+		ret = -EFAULT;
+		goto out;
+	}
+
+	if (lua_getfield(L, -1, "driver") != LUA_TTABLE) {
+		pr_err("Driver table is not specified");
+		ret = -EFAULT;
+		goto out;
+	}
+
+	if (lua_getfield(L, -1, "probe") != LUA_TFUNCTION) {
+		pr_warn("Probe function is not specified");
+		ret = 0;
```

Comment:
does it make sense to not have this specified? it looks to me we should return error on this..

Reply:
so, there's no point on the warn..